code/modules/library/computers/remote_gallery_access.dm

```diff
@@ -136,9 +136,23 @@
 /obj/machinery/computer/library/checkout/remote_gallery/make_external_book(var/datum/cachedbook/newbook)
 	if(!newbook)
 		return
-	var/obj/item/mounted/frame/painting/custom/C = new(get_turf(src))
+
+	var/obj/item/mounted/frame/painting/custom/C
+	var/datum/custom_painting/painting_data = json2painting(newbook.content, newbook.title, newbook.author, newbook.description)
+
+	//pick a canvas that fits the bitmap size.
+	if (painting_data.bitmap_width == 24)
```

Comment:
given that json2painting does not set bitmap_width or bitmap_height, I see no reason this should work

Reply:
well shit, back to the drawing board

Reply:
Actually, wait
Looking at [custom_painting_datum.dm](https://github.com/vgstation-coders/vgstation13/blob/07cfc50475dfaeab5e7c2e2ba19ae2ba3a566c30/code/modules/html_interface/paintTool/custom_painting_datum.dm):

I had to test it out but /proc/painting2json does produce a json with the width/height, which as far as I can tell is what gets saved to the DB
https://github.com/vgstation-coders/vgstation13/blob/07cfc50475dfaeab5e7c2e2ba19ae2ba3a566c30/code/modules/html_interface/paintTool/custom_painting_datum.dm#L470-L480

And json2painting does pass along the width and height from the json when creating the painting_data:
https://github.com/vgstation-coders/vgstation13/blob/07cfc50475dfaeab5e7c2e2ba19ae2ba3a566c30/code/modules/html_interface/paintTool/custom_painting_datum.dm#L483-L485

Am I missing something? The width and height should be informed as far as I can tell. 
Hell, thinking about it, if bitmap_width and bitmap_height weren't set, the painting wouldn't render at all.

Reply:
Nervere helped me out with setting up a DB on local. Way easier than I was expecting, I don't know why I've been scared of doing so for so long.

Anyway, fix seems to be working fine, no changed needed:
![Capture](https://github.com/vgstation-coders/vgstation13/assets/78439377/dab2f327-506d-401f-b2b5-37f74fe30757)

Let's call it 25% tested until I check the other sizes, but it's looking good so far.

Reply:
100% tested
![Capture](https://github.com/vgstation-coders/vgstation13/assets/78439377/5bb6bd89-0119-470c-85b4-2265bde1a800)

Reply:
Interesting, it doesn't seem to set it but perhaps it does elsewhere, spaghetti

Thanks for testing this